### ISSUE TITLE
Review effect of `bundled` on unique paths

### ISSUE THREAD

**Issue by bradcray** [chapel-lang/chapel#16790]
It seems that one effect of the `bundled` change from the other week (#16540) is that the hwloc-specific part of the path now simply says `bundled` where it used to say `hwloc`.  This is probably still sufficiently unique to not cause conflicts (assuming that we use the same components in the same order all the time), but to keep the path comprehensible to human readers, it seems we should make it `hwloc-bundled` (or `hwloc-system`, presumably).

**Comment by bradcray** [comment]
Tagging @daviditen on this, though others could grab it as well.  I think it'd be good to fix this soon (seems like it should be easy).

**Comment by aconsroe-hpe** [comment]
Combing issues, is this still high priority? It looks like hwloc and re2 both have a `bundled` part of the unique path without a prefix.
 * lib
 * linux64
 * gnu
 * x86_64
 * arch-native
 * loc-flat
 * comm-none
 * tasks-qthreads
 * tmr-generic
 * unwind-none
 * mem-jemalloc
 * atomics-cstdlib
 * bundled  `<-- this is hwloc`
 * bundled  `<-- this is re2`
 * fs-none
 * lib_pic-none
 * sanitizers-none
 * libchpl.a

I think the fix is:
```diff
diff --git a/util/chplenv/printchplenv.py b/util/chplenv/printchplenv.py
index 0323d6f3d4..2a4da57b6c 100755
--- a/util/chplenv/printchplenv.py
+++ b/util/chplenv/printchplenv.py
@@ -97,8 +97,8 @@ CHPL_ENVS = [
     ChapelEnv('CHPL_ATOMICS', RUNTIME | LAUNCHER | DEFAULT, 'atomics'),
     ChapelEnv('  CHPL_NETWORK_ATOMICS', INTERNAL | DEFAULT),
     ChapelEnv('CHPL_GMP', INTERNAL | DEFAULT, 'gmp'),
-    ChapelEnv('CHPL_HWLOC', RUNTIME | DEFAULT),
-    ChapelEnv('CHPL_RE2', RUNTIME | DEFAULT),
+    ChapelEnv('CHPL_HWLOC', RUNTIME | DEFAULT, 'hwloc'),
+    ChapelEnv('CHPL_RE2', RUNTIME | DEFAULT, 're2'),
     ChapelEnv('CHPL_LLVM', COMPILER | DEFAULT, 'llvm'),
     ChapelEnv('  CHPL_LLVM_CONFIG', COMPILER | NOPATH),
     ChapelEnv('  CHPL_LLVM_CLANG_C', INTERNAL),
```
And I think I remember @jhh67 wanting some prefixes on these paths for the printchplbuild tool. Are there others that I'm missing that are unnamed?

**Comment by bradcray** [comment]
It looks like I called this high priority from the perspective of "this looks easy and we should hit it while it's hot" and a bit of uneasiness ("could this bite us?").  It hasn't bitten us, so it probably isn't high priority and is no longer hot, so probably the high priority label should be removed, but I like that you're looking at closing it all the same.

> I'm tempted to rename the arch prefix to cpu and add the two prefixes for hwloc and re2

I'd support this and would be similarly tempted.

**Comment by aconsroe-hpe** [comment]
Merged #18866 ready to close?

**Comment by bradcray** [comment]
Yep, thank you!
